### ISSUE TITLE
records.json not present on the errand VM using Azure

### ISSUE THREAD

**Issue by srm09** [cloudfoundry/bosh#2182]
**Describe the bug**
`bosh-dns` was unable to resolve an alias for another VM in the deployment.
On inspection, the Bosh-DNS logs showed that it was unable to open records file at /var/vcap/instance/dns/records.json.
The file was not present at the location mentioned above.

**To Reproduce**
This has been happening when we run an errand on a separate VM while creating a PKS cluster. It is a part of our CI pipeline, and we can share the bosh debug logs for the deployment with the VM
  
**Expected behavior**
We would expect a records.json to be present at /var/vcap/instance/dns/records.json.

**Logs**
Logs are always helpful! Add logs to help explain your problem.

**Versions (please complete the following information):**
 - Infrastructure: Azure
 - BOSH version [268.2.2]
 - BOSH CLI version [5.4.0]
 - Stemcell version [ubuntu-xenial/170.13]
 - BOSH DNS version [1.10.0]

**Deployment info:**
NA

**Additional context**
We have been able to reproduce this issue consistently in a couple of our test environments.

**Comment by srm09** [comment]
Similar to the closed issue #2103

**Comment by srm09** [comment]
@nehagjain15 can you add any more relevant info that I might have missed.

**Comment by voelzmo** [comment]
@srm09 @nehagjain15 could you paste the relevant parts of how the errand and dns addon is configured in the manifest, please?

**Comment by voelzmo** [comment]
Here is what worked for me with the dummy-boshrelease from https://github.com/pivotal-cf-experimental/dummy-boshrelease on a bosh-lite setup with bosh-deployment.

```
---
name: dummy

releases:
- name: dummy
  version: create
  url: file:///Users/d058546/workspace/dummy-boshrelease

stemcells:
- alias: ubuntu
  os: ubuntu-xenial
  version: latest

instance_groups:
- name: dummy
  azs: [z1]
  instances: 2
  vm_type: default
  stemcell: ubuntu
  networks:
  - name: private
  jobs:
  - name: dummy_with_properties
    release: dummy
    properties:
      dummy_with_properties:
        echo_value: "hello"
- name: dummy-errand
  azs: [z1]
  instances: 1
  lifecycle: errand
  vm_type: default
  networks:
  - name: private
  stemcell: ubuntu
  jobs:
  - name: dummy_errand
    release: dummy

update:
  canaries: 0
  max_in_flight: 33
  serial: false
  canary_watch_time: 1000-60000
  update_watch_time: 1000-60000

features:
  use_tmpfs_config: true

```

And then executing 
```
# bosh -d dummy run-errand dummy_errand --keepalive
# bosh -d dummy ssh dummy-errand/0
# cat /var/vcap/instance/dns/records.json
```
showed that `records.json` is there and contains entries for the other VMs.

Director info
```
$ bosh env
Using environment '192.168.50.6' as client 'admin'

Name               bosh-lite
UUID               abb3a6a3-838e-41f2-96be-a9ea0efc3338
Version            269.0.1 (00000000)
Director Stemcell  ubuntu-xenial/315.13
CPI                warden_cpi
Features           compiled_package_cache: disabled
                   config_server: enabled
                   local_dns: enabled
                   power_dns: disabled
                   snapshots: disabled
User               admin
```

Stemcell used is 315.22

**Comment by srm09** [comment]
@voelzmo It is an intermittent issue that we are facing only in some of our azure environments.

**Comment by h4xnoodle** [comment]
@srm09 are there any debug logs available for deploys which produced instances without the records.json file? Additionally, are there any logs in `/var/vcap/bosh/log/current` that show any errors with respect to getting things from the blobstore or opening the records.json file?

In looking as to how the records.json is placed and populated, the director prepares the information, sends to the blobstore, and then the agent picks it up on the first `sync_dns` call made. The records.json file is written when the blobstore object can be found, read, and written again. Are there any logs around that suggest that the director didn't upload to the blobstore?

We'll also find you on slack to ask more questions.

**Comment by belinda-liu** [comment]
Summary from slack conversations: We found that this issue was primarily caused by the director_sync_dns not waiting for database migrations to finish. This was fixed in v269.0 but hadn't been backported to older versions. We've since backported that fix, and the latest releases on those LTS lines should address this issue.
